### DISCUSSION TITLE
Open cypress test runner UI within code-server

### DISCUSSION THREAD

**Discussion by TheZoker** [coder/code-server#5180] [Q&A]
Hi there,

I run code-server within a docker container (using docker compose) behind a nginx proxy. Within the code server I have a project open that includes cypress tests. 
Running them in headless mode works perfectly, but I want to use the UI (aka `cypress open`) as well. 

Is there any easy way to archive that? 

Thanks!

**Comment by jsjoeio** [comment]
Hmm...I've used Cypress before but it's been a while. I wonder if the nginx proxy in your setup is causing issues. What happens when you use the UI? Does it log any errors?

**Reply by TheZoker** [reply]
I think the issue is, that cypress requires some sort of desktop to open the chrome/firefox/electron browser and this is missing within docker / code-server. I thought there would maybe be a workaround that allows to run the browser on a specific port or something, but could not find any solution. I hoped that anybody had similar issues and found an easy solution for this.

**Reply by jsjoeio** [reply]
Ah, I see. Have you asked in the Cypress repo/discussions?
